x-pack/plugins/security/server/routes/users/change_password.ts

```diff
@@ -80,6 +81,9 @@ export function defineChangeUserPasswordRoutes({
       // session and in such cases we shouldn't create a new one.
       if (isUserChangingOwnPassword && currentSession) {
         try {
+          // Even though user is still the same, password change warrants a new session.
+          await session.clear(request);
```

Comment:
__note:__ maybe we even should do a logout here (to invalidate tokens when `token` provider is used), will think.